src/target/samx5x.c

```diff
@@ -474,11 +474,16 @@ static bool samx5x_flash_erase(target_flash_s *f, target_addr_t addr, size_t len
 		return false;
 	}
 
+	bool is_first_section = true;
+
 	for (size_t offset = 0; offset < len; offset += f->blocksize) {
 		target_mem_write32(t, SAMX5X_NVMC_ADDRESS, addr + offset);
 
-		/* Unlock */
-		samx5x_unlock_current_address(t);
+		/* If we're about to touch a new flash region, unlock it. */
+		if (is_first_section || ((offset % lock_region_size) == 0)) {
```

Comment:
The parens around the second half of the check are not required, please drop them like as in:
```suggestion
		if (is_first_section || (offset % lock_region_size) == 0) {
```

They suppress the GCC warning for accidental assignment which we want here.